update matrix.comp and run_test

JessicaMeixner-NOAA opened the pull request:
This is so that when we run matrix.comp we get clean comparisons, this required making sure we remove the ounfmeta.inp input in run_test (didn't test this, just deleted it from directories) and ran the comparison for ww3_tp2.14 from this branch twice so that we can easily check if there are issues in the future (instead of seeing that 204 files change as it would be easy to skip over something important that way).  

@mickaelaccensi please make sure there are not files that we should be comparing by removing all of the oasis/toy directories.

mickaelaccensi commented:
I've tested your new version of matrix.comp. It works and produces the same matrix outputs as I had before. However I have some comments

A remaining problem is the non-existing symbolic links like ounfmeta.inp which creates problems : 
`grep: ounfmeta.inp: No such file or directory`

I recommend to add the error checker on the first line :  
`#!/bin/bash -e `

Also, this line is not considered correct in new bash version : 
`cases=$(find . -maxdepth 1 -type d -name "work*" | grep -v work_oasis)`
it should be replaced by : 
`cases=$(find . -maxdepth 1 -type d -name "work*" -and ! -name "work_oasis*")`

Another correction would be to replace this section since oasis is now compiled inside each work sub-directory :
```
        cases=$(find . -maxdepth 1 -type d -name "work*" | grep -v work_oasis)
        # skip to next tst if no work directory
        echo 'run : ' $cases
        if [ -z "$cases" ] ; then
          echo "skip"
          continue
        fi
```
and replace it by : 
`cases=$(find . -maxdepth 1 -type d -name "work*")`

JessicaMeixner-NOAA commented:
For the 
grep: ounfmeta.inp: No such file or directory

my hope was that the update to the run_test would actually fix that.  Perhaps this needs more testing?   If you remove ounfmeta.inp from your directories (which the run_test should now do for you moving forward), do you still have this error? 

I added the -e in matrix.comp 

I think I updated the cases issue, but let me know if you don't think that's correct or it's not working.

mickaelaccensi commented:
ok I didn't rerun the test so that's why I still have the ounfmeta.inp.
I've tested successfully the new version
